cpg-analysis/src/main/kotlin/de/fraunhofer/aisec/cpg/query/FlowQueries.kt

```diff
@@ -55,26 +55,26 @@ fun FulfilledAndFailedPaths.toQueryTree(
         SinglePathResult(
```

Comment:
Can we directly extend this `QueryTree` implementation to add the assumptions? Can we somehow unify it with the `NodePath` class?

Reply:
I think that would need some larger rework where everywhere we are currently returning NodePath, we return SinglePathResult, or we make NodePath itself a QueryTree and swap out `SinglePathResults` for it. But I would leave that for the future.